### PR TITLE
add BackfillPolicy to the public API

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/backfill_policy.py [hunk 1]
```diff
@@ -21,35 +21,31 @@ class BackfillPolicy(
...
+      be backfilled in 10 dagster runs; each dagster run will backfill 10 partitions.
 
     - If an asset has 100 partitions, and the `max_partitions_per_run` is set to 11, then it will
-    be backfilled in 10 dagster runs, the first 9 dagster runs will backfill 11 partitions, and the last one run
-    will backfill the remaining 9 partitions.
-
-    For an asset that is not partitioned, backfill policy should not be assigned to it since we can only backfill
-    such asset as a whole.
+      be backfilled in 10 dagster runs; the first 9 dagster runs will backfill 11 partitions, and the
+      last one run will backfill the remaining 9 partitions.
```
Comment (erinkcochran87) [inline python_modules/dagster/dagster/_core/definitions/backfill_policy.py:42]: `9` should be `nine`, as numbers under 10 should be spelled out

**Review by erinkcochran87 — COMMENTED** [review]
Overall I think this looks fine! I did notice there are lots of instances of `dagster` in the description when it should be `Dagster` - if you can do a find/replace for that, that'd be great. I'll leave it to Claire to give the final stamp.

**Comment by sryza** [comment]
@erinkcochran87 thanks for taking a look.  I updated it to avoid the lowercase uses of "dagster". I'm on board with general guidance for spelling out numbers under 10, but I think in this case it would look weird to mix spelled out numbers with non-spelled out numbers in a paragraph that's walking through a calculation that involves both

**Review by clairelin135 — APPROVED** [review]
LGTM

**Comment by sryza** [comment]
@erinkcochran87 I'm going to merge this because I got the approve from claire and you didn't request changes and want to get it in before I leave for PTO later today. Happy to continue discussion the numeral spelling issue and potentially do a followup to make changes based on that discussion.

**Pull request merged**